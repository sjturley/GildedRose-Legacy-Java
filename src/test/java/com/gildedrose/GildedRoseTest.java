package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.Test;

import java.util.Arrays;

public class GildedRoseTest {

    @Test
    public void approveGildedRose() {
        String[] nameChoices = {"my item", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"};
        Integer[] sellInChoices = {-1, 0, 11};
        Integer[] qualityChoices = {0, 49, 50};
        CombinationApprovals.verifyAllCombinations(this::updateQuality, nameChoices, sellInChoices, qualityChoices);
    }

    private String updateQuality(String name, int sellIn, int quality) {
        GildedRose gildedRose = new GildedRose() {
            @Override
            public Item[] getItems() {
                return new Item[]{new Item(name, sellIn, quality)};
            }
        };
        gildedRose.updateQuality();

        return Arrays.toString(gildedRose.items);
    }
}
