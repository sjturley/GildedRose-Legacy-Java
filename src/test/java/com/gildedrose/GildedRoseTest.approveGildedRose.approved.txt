[my item, -1, 0] => [Item{name='my item', sellIn=-2, quality=0}] 
[my item, -1, 49] => [Item{name='my item', sellIn=-2, quality=47}] 
[my item, -1, 50] => [Item{name='my item', sellIn=-2, quality=48}] 
[my item, 0, 0] => [Item{name='my item', sellIn=-1, quality=0}] 
[my item, 0, 49] => [Item{name='my item', sellIn=-1, quality=47}] 
[my item, 0, 50] => [Item{name='my item', sellIn=-1, quality=48}] 
[my item, 11, 0] => [Item{name='my item', sellIn=10, quality=0}] 
[my item, 11, 49] => [Item{name='my item', sellIn=10, quality=48}] 
[my item, 11, 50] => [Item{name='my item', sellIn=10, quality=49}] 
[Aged Brie, -1, 0] => [Item{name='Aged Brie', sellIn=-2, quality=2}] 
[Aged Brie, -1, 49] => [Item{name='Aged Brie', sellIn=-2, quality=50}] 
[Aged Brie, -1, 50] => [Item{name='Aged Brie', sellIn=-2, quality=50}] 
[Aged Brie, 0, 0] => [Item{name='Aged Brie', sellIn=-1, quality=2}] 
[Aged Brie, 0, 49] => [Item{name='Aged Brie', sellIn=-1, quality=50}] 
[Aged Brie, 0, 50] => [Item{name='Aged Brie', sellIn=-1, quality=50}] 
[Aged Brie, 11, 0] => [Item{name='Aged Brie', sellIn=10, quality=1}] 
[Aged Brie, 11, 49] => [Item{name='Aged Brie', sellIn=10, quality=50}] 
[Aged Brie, 11, 50] => [Item{name='Aged Brie', sellIn=10, quality=50}] 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => [Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0}] 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => [Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0}] 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => [Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-2, quality=0}] 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => [Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0}] 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => [Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0}] 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => [Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=-1, quality=0}] 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => [Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=1}] 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => [Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=50}] 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => [Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=10, quality=50}] 
[Sulfuras, Hand of Ragnaros, -1, 0] => [Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=0}] 
[Sulfuras, Hand of Ragnaros, -1, 49] => [Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=49}] 
[Sulfuras, Hand of Ragnaros, -1, 50] => [Item{name='Sulfuras, Hand of Ragnaros', sellIn=-1, quality=50}] 
[Sulfuras, Hand of Ragnaros, 0, 0] => [Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=0}] 
[Sulfuras, Hand of Ragnaros, 0, 49] => [Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=49}] 
[Sulfuras, Hand of Ragnaros, 0, 50] => [Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=50}] 
[Sulfuras, Hand of Ragnaros, 11, 0] => [Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=0}] 
[Sulfuras, Hand of Ragnaros, 11, 49] => [Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=49}] 
[Sulfuras, Hand of Ragnaros, 11, 50] => [Item{name='Sulfuras, Hand of Ragnaros', sellIn=11, quality=50}] 
